src/core/lib/iomgr/ev_apple.cc

```diff
@@ -0,0 +1,354 @@
+                                      dispatch_queue_t dispatch_queue) {
+  grpc_apple_register_write_stream_impl(write_stream, dispatch_queue);
+}
+
+/// Drive the run loop in a global singleton thread until the global run loop is
+/// shutdown.
+static void GlobalRunLoopFunc(void* arg) {
+  grpc_core::ReleasableMutexLock lock(&gGlobalRunLoopContext->mu);
+  gGlobalRunLoopContext->run_loop = CFRunLoopGetCurrent();
+  gGlobalRunLoopContext->init_cv.Signal();
+
+  while (!gGlobalRunLoopContext->is_shutdown) {
+    // CFRunLoopRun() will return immediately if no stream is registered on it.
+    // So we wait on a conditional variable until a stream is registered;
+    // otherwise we'll be running a spinning loop.
+    while (!gGlobalRunLoopContext->input_source_registered) {
+      gGlobalRunLoopContext->input_source_cv.Wait(&gGlobalRunLoopContext->mu);
+    }
+    gGlobalRunLoopContext->input_source_registered = false;
+    lock.Unlock();
+    CFRunLoopRun();
+    lock.Lock();
+  }
+  lock.Unlock();
+}
+
+// pollset implementation
+
+static void pollset_global_init(void) {
+  gGlobalRunLoopContext = new GlobalRunLoopContext;
+
+  grpc_apple_register_read_stream_impl =
+      grpc_apple_register_read_stream_run_loop;
+  grpc_apple_register_write_stream_impl =
+      grpc_apple_register_write_stream_run_loop;
+
+  grpc_core::MutexLock lock(&gGlobalRunLoopContext->mu);
+  gGlobalRunLoopThread =
+      new grpc_core::Thread("apple_ev", GlobalRunLoopFunc, nullptr);
+  gGlobalRunLoopThread->Start();
+  while (gGlobalRunLoopContext->run_loop == NULL)
+    gGlobalRunLoopContext->init_cv.Wait(&gGlobalRunLoopContext->mu);
+}
+
+static void pollset_global_shutdown(void) {
+  {
+    grpc_core::MutexLock lock(&gGlobalRunLoopContext->mu);
+    gGlobalRunLoopContext->is_shutdown = true;
+    CFRunLoopStop(gGlobalRunLoopContext->run_loop);
+  }
+  gGlobalRunLoopThread->Join();
+  delete gGlobalRunLoopThread;
+  delete gGlobalRunLoopContext;
+}
+
+/// The caller must acquire the lock GrpcApplePollset.mu before calling this
+/// function. The lock may be temporarily released when waiting on the condition
+/// variable but will be re-acquired before the function returns.
+///
+/// The Apple pollset simply waits on a condition variable until it is kicked.
+/// The network events are handled in the global run loop thread. Processing of
+/// these events will eventually trigger the kick.
+static grpc_error* pollset_work(grpc_pollset* pollset,
+                                grpc_pollset_worker** worker,
+                                grpc_millis deadline) {
+  GRPC_POLLING_TRACE("pollset work: %p, worker: %p, deadline: %" PRIu64,
+                     pollset, worker, deadline);
+  GrpcApplePollset* apple_pollset =
+      reinterpret_cast<GrpcApplePollset*>(pollset);
+  GrpcAppleWorker actual_worker;
+  if (worker) {
+    *worker = reinterpret_cast<grpc_pollset_worker*>(&actual_worker);
+  }
+
+  if (apple_pollset->outstanding_kick) {
```

Comment:
If this is the same as the [kicked_without_poller](https://github.com/grpc/grpc/blob/master/src/core/lib/iomgr/ev_epollex_linux.cc#L1129) in ev_epollex_linux, I suggest using the same name for easy understanding.